src/merge.ts

```diff
@@ -25,16 +25,20 @@ export function merge(
 
   const mergedOptions = mergeDeep(defaults || {}, options) as EndpointDefaults;
 
-  // mediaType.previews arrays are merged, instead of overwritten
-  if (defaults && defaults.mediaType.previews.length) {
-    mergedOptions.mediaType.previews = defaults.mediaType.previews
-      .filter((preview) => !mergedOptions.mediaType.previews.includes(preview))
-      .concat(mergedOptions.mediaType.previews);
+  if (options.url === "/graphql") {
+    // mediaType.previews arrays are merged, instead of overwritten
+    if (defaults && defaults.mediaType.previews!.length) {
```

Comment:
`.previews` is removed from `src/defaults.ts`, I don't think we can assume that it's set.

```suggestion
    if (defaults && defaults.mediaType.previews?.length) {
```